### ISSUE TITLE
Slow performance on large channels

### ISSUE THREAD

**Issue by scj643** [magne4000/quassel-webserver#238]
I'm noticing really slow performance when am in large channels like #python on freenode.

**Comment by magne4000** [comment]
Yep, but quassel protocol is not really optimized for such large channels, and using it through web (or mobile) is sadly something that I can't really fix.

**Comment by fish-face** [comment]
I am not sure this has anything to do with the quassel protocol. I am on some huge channels through bitlbee, and just typing in the input box or switch to the channel is excruciatingly slow. Compare that to a medium channel and it's alright, but not great. (This is on a fairly old work computer.) This should not have to do anything protocol related, right?

**Comment by fish-face** [comment]
@magne4000 I ran the profiler in Firefox in a channel with a couple of thousand users. Every single keypress, maybe more often, is causing a very long layout recalculation. I suspect something is incorrectly trying to re-layout the users list every single keypress, but it's not obvious from the profiles (most stuff just seems to be angular.js functions). If you need help getting profiles let me know!
